### PR TITLE
Added facility for tracking and exposing basic database-level stats

### PR DESCRIPTION
**Pull request by oxtoacart**
Part 1 of #12.

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		delete(followerStats, followerID)
+		partitionStats[fs.Partition].NumFollowers--
+	}
+}
+
+// QueuedForFollower records how many measurements are queued for a given Follower
+func QueuedForFollower(followerID int, queued int) {
+	mx.Lock()
+	defer mx.Unlock()
+	followerStats[followerID].Queued = queued
```
Comment (forkner) [inline metrics/metrics.go]: It looks okay by eyeball --  but could `followerStats[followerID]` be `nil` here if it was in a race for the lock and `FollowerFailed` wins?

It looks like markFailed can only be triggered from a read that starts after processing that follower, but it's a bit subtle, so just wanted to check if that was the expectation.

**Comment by forkner** [comment]
@oxtoacart looks okay to me, not sure how to interpret the CI results.  master also seems to be having issues.

File: metrics/metrics.go (hunk 1 above)
Comment (oxtoacart) [reply metrics/metrics.go]: Yes, that is the expectation. Great question!

**Comment by oxtoacart** [comment]
Thanks @forkner!

**Pull request merged**